tests/python/unittest/test_im2rec.py

```diff
@@ -0,0 +1,112 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# pylint: skip-file
+import mxnet as mx
+import mxnet.ndarray as nd
+from mxnet.test_utils import *
+from mxnet.base import MXNetError
+import numpy as np
+import os
+import tarfile
+import sys
+from common import assertRaises
+import unittest
+
+def test_im2rec():
+    if not os.path.isdir("data/test_im2rec/"):
+        os.makedirs('data/test_im2rec/')
+    tar_filename = download('http://data.mxnet.io/data/test_images.tar.gz',
+                            dirname='data/test_im2rec/')
+    with tarfile.open(tar_filename, "r:gz") as gf:
+        gf.extractall("data/test_im2rec/")
+    filenames = next(os.walk("data/test_im2rec/test_images"))[2]
+    filenames = [os.path.abspath(os.path.join("data/test_im2rec/test_images",i))
+                 for i in filenames]
+    # creating list file from test image data
+    lst_filename = os.path.abspath("data/test_im2rec/img.lst")
+    output_path = os.path.abspath("data/test_im2rec/")
+    with open(lst_filename, 'w') as img_lst:
+        for i,fpath in enumerate(filenames):
+            img_lst.write(str(i) + "\t" + str(i) + "\t" + fpath + "\n")
+    
+    def test_validate_params():
+        try:
+            mx.io.im2rec._validate_params("notafile", output_path)
```

Comment:
You can use nose to check that throws. And put each on its own function with that decorator